caffe2/CMakeLists.txt

```
@@ -1210,9 +1239,30 @@ endif()
 # Pass USE_DISTRIBUTED to torch_cpu, as some codes in jit/pickler.cpp and
 # jit/unpickler.cpp need to be compiled only when USE_DISTRIBUTED is set
 if(USE_DISTRIBUTED)
-  target_compile_definitions(torch_cpu PRIVATE
-    USE_DISTRIBUTED
-  )
+  target_include_directories(torch_cpu PUBLIC ${TORCH_SRC_DIR}/lib)
+  target_compile_definitions(torch_cpu PRIVATE USE_DISTRIBUTED)
+  if(USE_GLOO AND USE_C10D_GLOO)
+    target_compile_definitions(torch_cpu PUBLIC USE_C10D_GLOO)
+  endif()
+  if(USE_NCCL AND USE_C10D_NCCL)
+    if(USE_ROCM)
+      target_compile_definitions(torch_hip PUBLIC USE_C10D_NCCL)
+    elseif(USE_CUDA)
```

Nit: `USE_NCCL` implies that either USE_CUDA or USE_ROCM is defined
```suggestion
    else()
```